src/components/organisms/EntityList/EntityCreationModal/TestCreationForm/TestCreationForm.tsx

```diff
@@ -0,0 +1,183 @@
+  getSourceFieldValue,
+  getSourcePayload,
+  remapTestSources,
+  testSourceBaseOptions,
+} from '@utils/sources';
+
+import {useAddTestMutation} from '@services/tests';
+
+import {LabelsWrapper, StyledFormSpace} from '../CreationModal.styled';
+
+type TestCreationFormProps = {
+  form: FormInstance;
+  onSuccess: (res: AddTestPayload) => void;
+  onFail: (err: any) => void;
+  testSources: SourceWithRepository[];
+  executors: Executor[];
+};
+
+const additionalFields: {[key: string]: React.FC<any>} = {
+  git: GitCreationFormFields,
+  'file-uri': FileContentFields,
+  custom: CustomCreationFormFields,
+  string: StringContentFields,
+};
+
+const TestCreationForm: React.FC<TestCreationFormProps> = props => {
+  const {form, testSources, executors, onSuccess, onFail} = props;
+
+  const remappedExecutors = remapExecutors(executors);
+  const remappedCustomTestSources = remapTestSources(testSources);
+
+  const [localLabels, setLocalLabels] = useState<readonly Option[]>([]);
+
+  const [addTest, {isLoading}] = useAddTestMutation();
+
+  const onSave = (values: any) => {
+    const {testSource, testType} = values;
+
+    const requestBody = {
+      name: values.name,
+      type: testType,
+      labels: decomposeLabels(localLabels),
+      content: getSourcePayload(values, testSources),
+      ...getCustomSourceField(testSource),
+    };
+
+    return addTest(requestBody)
+      .then(res => {
+        onSuccess(res);
+      })
+      .catch(err => {
+        onFail(err);
+      });
+  };
+
+  return (
+    <Form form={form} layout="vertical" name="test-creation" onFinish={onSave} style={{flex: 1}} labelAlign="right">
+      <StyledFormSpace size={24} direction="vertical">
+        <Text className="regular big">Test details</Text>
+        <FormItem
+          name="name"
+          label="Name"
+          rules={[required, k8sResourceNamePattern, k8sResourceNameMaxLength]}
+          required
+        >
+          <Input placeholder="e.g.: my-test" />
+        </FormItem>
+        <FormItem name="testType" label="Type" rules={[required]} required>
+          <Select placeholder="Select from available executors..." showSearch options={remappedExecutors} allowClear />
+        </FormItem>
+        <FormItem
+          noStyle
+          shouldUpdate={(prevValues, currentValues) => prevValues.testSource !== currentValues.testSource}
+        >
+          {({getFieldValue}) => {
+            const selectedExecutor = executors.find((executor: Executor) =>
+              executor.executor?.types?.includes(getFieldValue('testType'))
+            );
+
+            if (!selectedExecutor) {
+              return null;
+            }
+
+            const options = [
+              ...remappedCustomTestSources,
+              ...testSourceBaseOptions.filter(
+                option =>
+                  selectedExecutor.executor?.contentTypes?.includes(String(option.value)) ||
+                  !selectedExecutor.executor?.contentTypes?.length
```

Comment:
Is that correct that when the executor doesn't specify types it should allow any? Makes sense for me, but I'd like to confirm.

Reply:
Yes, it is